app/Services/UserService.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rules\Password;
+
+class UserService
+{
+    /**
+     * Change the user password.
+     *
+     * @param User  $user
+     * @param array $input
+     *
+     * @return void
+     */
+    public function changePassword(User $user, array $input): void
+    {
+        // Validate input
+        Validator::make($input, [
+            'current_password' => [
+                'required',
+                'string',
+            ],
+            'new_password' => [
+                'required',
+                'string',
+                Password::defaults(),
+            ],
+        ])->after(function ($validator) use ($user, $input) {
+            if (!isset($input['current_password']) || !Hash::check($input['current_password'], $user->password)) {
+                $validator->errors()->add('current_password', __('api.ERROR.CURRENT_PASSWORD_DOES_NOT_MATCH'));
+            }
+        })->validate();
```

Comment:
This can be simplified with https://laravel.com/docs/9.x/validation#rule-current-password . 
- Alternatively, I can open a new PR with this suggestion, tight after this is merged, and also include E2E tests on the route.

```suggestion
        Validator::make($input, [
            'current_password' => [
                'required',
                'string',
                'current_password',
            ],
            'new_password' => [
                'required',
                'string',
                Rules\Password::defaults(),
            ],
        ])->validate();
```

Reply:
Good catch

Reply:
I also noticed that the pt_BR translations were missing, so I took the opportunity to add it